### ISSUE TITLE
face_recognition pipeline for multiple sources

### ISSUE THREAD

**Issue by shubham-shahh** [ageitgey/face_recognition#1274]
* face_recognition version: -
* Python version:3.7
* Operating System: ubuntu 18.04

### Description

Hi,
I am have implemented a face recognition pipeline with face_recognition library on jetson nano which fetched 7-10 fps from a single 1080p source, which is decent for a single source. but, to increase the number of sources I am planning to use this library with deepstream to process multiple streams.
please do let me know if you've implemented something similar.

**Comment by jirayu** [comment]
I have the need to have Python or similar that be able to recognize faces from a sequence of picture taken and write to a log file for payroll attendant.  Do you happen to have such solution ?  I came from Windows base with Delphi and C# stuff so really hard a hard time with integrating Python but to run it along and just use.

**Comment by shubham-shahh** [comment]
a sequence of pictures? is it a live feed? or images from a folder?
in either case, you can refer 
https://www.pyimagesearch.com/2018/06/18/face-recognition-with-opencv-python-and-deep-learning/
or
https://www.youtube.com/watch?v=535acCxjHCI

**Comment by FedericoLanzani** [comment]
Hi! I'm using a jetson nano with multiple video streams. 
I found it difficult to get deepstream working, so I decided to use opencv with gstreamer.
_Note_: Gstreamer is a very powerful library for managing video streams, deepstream is based on it.

Opencv has the possibility to run the video with Gstreamer, but this feature requires opencv to be compiled.
Luckly jetpack 4.4 comes with a pre-built version of opencv with gstreamer.

To use that you need to build a gstreamer pipeline and feed it to the cv2.VideoCapture method like that:
```
cam_source = 'v4l2src device=/dev/video0 ! video/x-raw, width=640, height=480, framerate=30/1 ! videoconvert !  appsink'
cv2.VideoCapture(cam_source)
```

**In conclusion**, learn how to use deepstream it's the best solution because you will have the best performance. Deepstream is also very powerful for a lot of tasks on the jetson. Otherwise, if you have to migrate your existing application that uses opencv to the jetson nano and you don't have a lot of time I suggest you to use opencv.

**Comment by shubham-shahh** [comment]
> Hi! I'm using a jetson nano with multiple video streams. 
> I found it difficult to get deepstream working, so I decided to use opencv with gstreamer.
> _Note_: Gstreamer is a very powerful library for managing video streams, deepstream is based on it.
> 
> Opencv has the possibility to run the video with Gstreamer, but this feature requires opencv to be compiled.
> Luckly jetpack 4.4 comes with a pre-built version of opencv with gstreamer.
> 
> To use that you need to build a gstreamer pipeline and feed it to the cv2.VideoCapture method like that:
> ```
> cam_source = 'v4l2src device=/dev/video0 ! video/x-raw, width=640, height=480, framerate=30/1 ! videoconvert !  appsink'
> cv2.VideoCapture(cam_source)
> ```
> 
> **In conclusion**, learn how to use deepstream it's the best solution because you will have the best performance. Deepstream is also very powerful for a lot of tasks on the jetson. Otherwise, if you have to migrate your existing application that uses opencv to the jetson nano and you don't have a lot of time I suggest you to use opencv.

Use [this](https://github.com/shubham-shahh/mtcnn_facenet_cpp_tensorRT/tree/develop) fork to use facenet with deepstream.

**Comment by LUOQING1994** [comment]
> > Hi! I'm using a jetson nano with multiple video streams.
> > I found it difficult to get deepstream working, so I decided to use opencv with gstreamer.
> > _Note_: Gstreamer is a very powerful library for managing video streams, deepstream is based on it.
> > Opencv has the possibility to run the video with Gstreamer, but this feature requires opencv to be compiled.
> > Luckly jetpack 4.4 comes with a pre-built version of opencv with gstreamer.
> > To use that you need to build a gstreamer pipeline and feed it to the cv2.VideoCapture method like that:
> > ```
> > cam_source = 'v4l2src device=/dev/video0 ! video/x-raw, width=640, height=480, framerate=30/1 ! videoconvert !  appsink'
> > cv2.VideoCapture(cam_source)
> > ```
> > 
> > 
> > **In conclusion**, learn how to use deepstream it's the best solution because you will have the best performance. Deepstream is also very powerful for a lot of tasks on the jetson. Otherwise, if you have to migrate your existing application that uses opencv to the jetson nano and you don't have a lot of time I suggest you to use opencv.
> 
> Use [this](https://github.com/shubham-shahh/mtcnn_facenet_cpp_tensorRT/tree/develop) fork to use facenet with deepstream.

Hi!
Thank you very much for your sharing. Through your fork, I have successfully run this program on Jetson nano and analyzed the data of four cameras at the same time. However, I found two problems:
1. The speed of using Yolo tiny to capture human face is only 5 ~ 6 FPS. In order to improve FPS, I want to convert fp32 RT model into int8 RT model. What should I do?
2. After the face is 3M away from the camera, the face can not be detected. What is the reason for this?

**Comment by shubham-shahh** [comment]
> > > Hi! I'm using a jetson nano with multiple video streams.
> > > I found it difficult to get deepstream working, so I decided to use opencv with gstreamer.
> > > _Note_: Gstreamer is a very powerful library for managing video streams, deepstream is based on it.
> > > Opencv has the possibility to run the video with Gstreamer, but this feature requires opencv to be compiled.
> > > Luckly jetpack 4.4 comes with a pre-built version of opencv with gstreamer.
> > > To use that you need to build a gstreamer pipeline and feed it to the cv2.VideoCapture method like that:
> > > ```
> > > cam_source = 'v4l2src device=/dev/video0 ! video/x-raw, width=640, height=480, framerate=30/1 ! videoconvert !  appsink'
> > > cv2.VideoCapture(cam_source)
> > > ```
> > > 
> > > 
> > > **In conclusion**, learn how to use deepstream it's the best solution because you will have the best performance. Deepstream is also very powerful for a lot of tasks on the jetson. Otherwise, if you have to migrate your existing application that uses opencv to the jetson nano and you don't have a lot of time I suggest you to use opencv.
> > 
> > 
> > Use [this](https://github.com/shubham-shahh/mtcnn_facenet_cpp_tensorRT/tree/develop) fork to use facenet with deepstream.
> 
> Hi!
> Thank you very much for your sharing. Through your fork, I have successfully run this program on Jetson nano and analyzed the data of four cameras at the same time. However, I found two problems:
> 
> 1. The speed of using Yolo tiny to capture human face is only 5 ~ 6 FPS. In order to improve FPS, I want to convert fp32 RT model into int8 RT model. What should I do?
> 2. After the face is 3M away from the camera, the face can not be detected. What is the reason for this?

Hi, int 8 quantization requires calibration file I would suggest using fp16 precision and on the other hand when you say 3M I am assuming you mean 3 meters, I cannot conclude the problem, please share a picture or a gif for clarity

Thanks and regards
